app/services/ExpensesRedirectService.scala

```diff
@@ -0,0 +1,189 @@
+        ConditionalRedirect(professionalSubscriptionsQuestion.isEmpty, ProfessionalFeesAndSubscriptionsExpensesController.show(taxYear)),
+        ConditionalRedirect(professionalSubscriptionsQuestion.contains(false), OtherEquipmentController.show(taxYear), hasPrior = Some(false)),
+        ConditionalRedirect(professionalSubscriptionsQuestion.contains(false), CheckEmploymentExpensesController.show(taxYear), hasPrior = Some(true))
+      )
+  }
+
+  def otherAllowancesRedirects(cya: ExpensesCYAModel, taxYear: Int): Seq[ConditionalRedirect] = {
+    val jobExpensesSectionFinished = cya.expenses.jobExpensesSectionFinished(taxYear)
+    val flatRateSectionFinished = cya.expenses.flatRateSectionFinished(taxYear)
+    val profSubscriptionsSectionFinished = cya.expenses.professionalSubscriptionsSectionFinished(taxYear)
+
+    commonExpensesRedirects(cya, taxYear) ++ Seq(
+      jobExpensesSectionFinished.map(ConditionalRedirect(_)),
+      flatRateSectionFinished.map(ConditionalRedirect(_)),
+      profSubscriptionsSectionFinished.map(ConditionalRedirect(_))
+    ).flatten
+  }
+
+  def otherAllowanceAmountRedirects(cya: ExpensesCYAModel, taxYear: Int): Seq[ConditionalRedirect] = {
+    val otherAllowanceQuestion = cya.expenses.otherAndCapitalAllowancesQuestion
+
+    otherAllowancesRedirects(cya, taxYear) ++
+      Seq(
+        ConditionalRedirect(otherAllowanceQuestion.isEmpty, OtherEquipmentController.show(taxYear)),
+        ConditionalRedirect(otherAllowanceQuestion.contains(false), CheckEmploymentExpensesController.show(taxYear), hasPrior = Some(false)),
+        ConditionalRedirect(otherAllowanceQuestion.contains(false), CheckEmploymentExpensesController.show(taxYear), hasPrior = Some(true))
+      )
+  }
+
+  def redirectBasedOnCurrentAnswers(taxYear: Int, data: Option[ExpensesUserData])
+                                   (cyaConditions: ExpensesCYAModel => Seq[ConditionalRedirect])
+                                   (block: ExpensesUserData => Future[Result]): Future[Result] = {
+    val redirect = calculateRedirect(taxYear, data, cyaConditions)
+
+    redirect match {
+      case Left(redirect) => Future.successful(redirect)
+      case Right(cya) => block(cya)
+    }
+  }
+
+  private def calculateRedirect(taxYear: Int, data: Option[ExpensesUserData],
+                                cyaConditions: ExpensesCYAModel => Seq[ConditionalRedirect]): Either[Result, ExpensesUserData] = {
+    data match {
+      case Some(cya) =>
+
+        val possibleRedirects = cyaConditions(cya.expensesCya)
+
+        val redirect = possibleRedirects.collectFirst {
+          case ConditionalRedirect(condition, result, Some(hasPriorExpenses)) if condition && hasPriorExpenses == cya.hasPriorExpenses => Redirect(result)
+          case ConditionalRedirect(condition, result, None) if condition => Redirect(result)
+        }
+
+        redirect match {
+          case Some(redirect) =>
+            logger.info(s"[RedirectService][calculateRedirect]" +
```

Comment:
log name can be updated to this file

Reply:
updated and got rid of those extra lines at the end